DeepLinkSDK/Router/DPLDeepLinkRouter.m

```diff
@@ -48,16 +52,31 @@ - (void)registerHandlerClass:(Class <DPLRouteHandler>)handlerClass forRoute:(NSS
     if (handlerClass && [route length]) {
         [self.routes addObject:route];
         [self.blocksByRoute removeObjectForKey:route];
+        [self.protocolsByRoute removeObjectForKey:route];
+        
         self.classesByRoute[route] = handlerClass;
     }
 }
 
+- (void)registerHandler:(id <DPLRouteHandlerProtocol>)handler forRoute:(NSString *)route {
+    
+    if (handler && [route length]) {
+        [self.routes addObject:route];
+        [self.blocksByRoute removeObjectForKey:route];
+        [self.classesByRoute removeObjectForKey:route];
+        
+        self.protocolsByRoute[route] = handler;
```

Comment:
I thought it might be confusing to new developers to see both the protocol and the class and wonder which one to use, so it may be good to call out the differences in the getting started documentation.

I've got one more change to submit- I added an optional asynchronous version of targetViewController, but wanted to finish integrating it into our app to make sure it's complete.


    typedef void (^DPLTargetViewControllerCompletionHandler)(UIViewController <DPLTargetViewController> * targetViewController);

    - (void)targetViewController:(DPLDeepLink *)deepLink completionHandler:(DPLTargetViewControllerCompletionHandler)completionHandler;